holoviews/operation/element.py

```diff
@@ -605,6 +605,53 @@ def _process(self, element, key=None):
         return sliced
 
 
+def pts_to_prestep(x, y):
+    steps = np.zeros((2, 2 * len(x) - 1))
+    steps[0, 0::2] = x
+    steps[0, 1::2] = steps[0, 0:-2:2]
+    steps[1:, 0::2] = y
+    steps[1:, 1::2] = steps[1:, 2::2]
+    return steps
+
+def pts_to_poststep(x, y):
+    steps = np.zeros((2, 2 * len(x) - 1))
+    steps[0, 0::2] = x
+    steps[0, 1::2] = steps[0, 2::2]
+    steps[1:, 0::2] = y
+    steps[1:, 1::2] = steps[1:, 0:-2:2]
+    return steps
+
+def pts_to_midstep(x, y):
+    steps = np.zeros((2, 2 * len(x)))
+    x = np.asanyarray(x)
+    steps[0, 1:-1:2] = steps[0, 2::2] = (x[:-1] + x[1:]) / 2
+    steps[0, 0], steps[0, -1] = x[0], x[-1]
+    steps[1:, 0::2] = y
+    steps[1:, 1::2] = steps[1:, 0::2]
+    return steps
+
+
+class step_curve(ElementOperation):
```

Comment:
I would consider generalizing this to an ``interpolate_curve``  that supports the same interpolation options as ``CurvePlot`` - currently this would just mean adding a 'linear' option that does nothing. I would also consider making the functions above into classmethods.